planner/core/logical_plan_builder.go

```diff
@@ -2438,6 +2438,7 @@ func (b *PlanBuilder) buildDataSource(ctx context.Context, tn *ast.TableName, as
 	if err != nil {
 		return nil, err
 	}
+	possiblePaths = b.filterPathByIsolationRead(possiblePaths)
```

Comment:
What if no path left?

Reply:
There is none proper plan found.

Reply:
Can we add a test cover this case? Or we should wait for your next pr?

Reply:
OK, I'll add a test to cover it.

Reply:
Done.